models/intermediate/int_marketo__lead.sql

```diff
@@ -0,0 +1,50 @@
+with leads as(
+    select * 
+    from {{ var('lead') }} --check the name
+
+), activity_merge_leads as (
+    select * 
+    from {{ var('activity_merge_leads') }}
+
+), unique_merges as (
+
+    select 
+        cast(lead_id as {{ dbt_utils.type_int() }}) as lead_id,
+        {{ fivetran_utils.string_agg('distinct merged_lead_id', "', '") }} as merged_into_lead_id
+
+    from activity_merge_leads
+    group by lead_id 
+
+--If you use activity_delete_lead tags this will be included, if not it will be ignored.
+{% if var('marketo__activity_delete_lead_enabled', True) %}
+), deleted_leads as (
+
+    select *
+    from {{ var('activity_delete_lead') }}
+    
+{% endif %}
+
+), joined as (
+
+    select 
+        leads.*,
+
+        --If you use activity_delete_lead tags this will be included, if not it will be ignored.
+        {% if var('marketo__activity_delete_lead_enabled', True) %}
+        case when deleted_leads.lead_id is not null then True else False end as is_deleted,
+        {% endif %}
```

Comment:
In this case we are completely removing the field if the user doesn't have the delete leads table. Would it be more impactful to instead pass a `null` record through this field if it isn't used instead of removing it altogether? What are your thoughts?

Reply:
Good point thanks. I have added an else to make it null if the `marketo__activity_delete_lead` table is disabled. I think this might signal the column could have data if the table was enabled.